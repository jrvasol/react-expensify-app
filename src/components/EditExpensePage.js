import React from 'react';
import { connect } from 'react-redux';
import { editExpense } from '../actions/expenses'
import { removeExpense } from '../actions/expenses'
import ExpenseForm from './ExpenseForm'

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }

    onRemoveExpense = () => {
        this.props.removeExpense({ id: this.props.expense.id }) 
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
                <ExpenseForm onSubmit={this.onSubmit} expense={this.props.expense}/>
                <button onClick={this.onRemoveExpense}>Remove</button>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}

const mapDispatchToProps = (dispatch, props) => {
    return {
        editExpense: (id, expense) => dispatch(editExpense(id, expense)),
        removeExpense: (id) => dispatch(removeExpense(id)) 
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);