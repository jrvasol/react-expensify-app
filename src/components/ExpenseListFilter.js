import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates'; 
import moment from 'moment';

export class ExpenseListFilters extends React.Component {
    state = {
        filterFocus: null
    }

    onDateChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    } 

    onFocusChange = (focus) => {
        this.setState({ filterFocus: focus })
    } 

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }

    onSortByChange = (e) => { 
        if(e.target.value === 'amount') {
            this.props.sortByAmount()
        } else if(e.target.value === 'date') {
            this.props.sortByDate(); 
        } 
    }

    render() {
        return (
            <div>
                <input type="text" 
                        onChange={this.onTextChange} 
                        value={this.props.filters.text} />
                <select value={this.props.filters.sortBy} 
                        onChange={this.onSortByChange}>
                    <option value="amount">Amount</option>
                    <option value="date">Date</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
                    startDateId="filterStartDate" // PropTypes.string.isRequired,
                    endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
                    endDateId="filterEndDate" // PropTypes.string.isRequired,
                    onDatesChange={this.onDateChange} // PropTypes.func.isRequired,
                    focusedInput={this.state.filterFocus} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                    onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
                    isOutsideRange={() => false}
                    numberOfMonths={1}
                    showClearDates={true}
                    showDefaultInputIcon={true}
                    regular={true}
                    />
            </div>
        )
    }   
}

const mapDispatchToProps = (dispatch) => {
    return {
        setTextFilter: value => dispatch(setTextFilter(value)),
        sortByAmount: () => dispatch(sortByAmount()),
        sortByDate: () => dispatch(sortByDate()),
        setStartDate: (date) => dispatch(setStartDate(date)),
        setEndDate: (date) => dispatch(setEndDate(date))
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);