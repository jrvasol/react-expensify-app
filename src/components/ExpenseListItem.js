import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({ id, description, amount, createdAt, dispatch, history }) => (
    <li className="expense"> 
        <Link to={'edit/' + id}>{ description }</Link>
        <p>{ amount / 100 } - { moment(createdAt).format() }</p>
    </li>
)

export default ExpenseListItem;