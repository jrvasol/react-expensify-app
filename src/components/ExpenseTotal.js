import React from 'react';
import {connect} from 'react-redux';
import getExpensesTotal from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses'
import numeral from 'numeral';


export const ExpenseTotal = (props) => (
    <div>
        <h1>Expense Total: {numeral(props.expensesTotal / 100).format('$0,0.00')}</h1>
        <p>Total Count: { props.expensesCount }</p>
    </div>
)

const mapStateToProps = (state) => {
    return {
        expensesTotal: getExpensesTotal(getVisibleExpenses(state.expenses, state.filters)),
        expensesCount: getVisibleExpenses(state.expenses, state.filters).length
    }
}

export default connect(mapStateToProps)(ExpenseTotal);