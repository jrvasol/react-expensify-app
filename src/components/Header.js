import React from 'react';
import {NavLink} from "react-router-dom";

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <nav>
            <ul>
                <li>
                    <NavLink to="/" activeClassName="is-active" exact>Home</NavLink>
                </li>
                <li>
                    <NavLink to="/create" activeClassName="is-active">Create</NavLink>
                </li>
            </ul>
        </nav>
    </header>
)

export default Header;