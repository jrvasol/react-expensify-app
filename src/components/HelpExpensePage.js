import React from 'react';

const HelpExpensePage = () => (
    <div>
        <h1>Help Page</h1>
    </div>
) 

export default HelpExpensePage;