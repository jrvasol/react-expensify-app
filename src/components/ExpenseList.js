import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem'
import getVisibleExpenses from '../selectors/expenses'

export const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {
            props.expenses.length === 0 ? (
                
                <p>No expenses available</p>

            ) : (
                <ul>
                    {
                        props.expenses.map((expense) => {
                            return <ExpenseListItem {...expense} key={expense.id} />
                        })
                    } 
                </ul>
            )
        }
    </div>
); 

const mapStateToProps = (state) => {
    return {
        expenses: getVisibleExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseList);