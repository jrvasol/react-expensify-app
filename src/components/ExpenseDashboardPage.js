import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilter'
import ExpenseTotal from './ExpenseTotal';

const ExpenseDashBoardPage = () => (
    <div>
        <ExpenseTotal/>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
)

export default ExpenseDashBoardPage