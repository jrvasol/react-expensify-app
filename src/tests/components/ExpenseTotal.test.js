import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseTotal} from '../../components/ExpenseTotal';
import expenses from '../fixtures/expenses';

test('should render ExpenseTotal with 1 value', () => {
    const wrapper = shallow(<ExpenseTotal expensesTotal={102103.2343} expensesCount={5}/>);
    expect(wrapper).toMatchSnapshot();
})