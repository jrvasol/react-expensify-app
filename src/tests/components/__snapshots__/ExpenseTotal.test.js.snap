// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseTotal with 1 value 1`] = `
<div>
  <h1>
    Expense Total: 
    $1,021.03
  </h1>
  <p>
    Total Count: 
    5
  </p>
</div>
`;
